File: php/src/CodingInterview/Application/Repository/PortfolioRepository.php
<?php

declare(strict_types=1);

namespace Folio\CodingInterview\Application\Repository;

use Folio\CodingInterview\Domain\Portfolio;

/**
 * 最適ポートフォリオリポジトリ。
 */
interface PortfolioRepository
{
    public function get(): Portfolio;

    public function update(Portfolio $portfolio): void;
}

File: scala/src/main/scala/folio/codinginterview/application/repository/PortfolioRepository.scala
package folio.codinginterview.application.repository

import folio.codinginterview.domain.Portfolio
import scala.concurrent.Future

/** 最適ポートフォリオリポジトリ。 */
trait PortfolioRepository {
  def get(): Future[Portfolio]
  def update(portfolio: Portfolio): Future[Unit]
}
